client/me/purchases/cancel-purchase/button.jsx

```diff
@@ -74,13 +143,35 @@ const CancelPurchaseButton = React.createClass( {
 		} );
 	},
 
+	handleSubmit( error, response ) {
+		if ( error ) {
+			notices.error( error.message );
+			return;
+		}
+
+		notices.success( response.message, { persistent: true } );
+
+		analytics.tracks.recordEvent(
+			'calypso_purchases_cancel_form_submit',
+			{ product_slug: this.props.purchase.productSlug }
+		);
+
+		page.redirect( paths.list() );
+	},
+
+	submitCancelPurchase() {
+		wpcom.cancelAndRefundPurchase( this.props.purchase.id, null, ( error, response ) => {
```

Comment:
Also, what do you think about moving this into the actions in `upgrades/actions/purchases.js`?